src/plugins/discover/public/application/components/results_action_bar/results_action_bar.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './results_action_bar.scss';
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { HitsCounter } from '../chart/hits_counter';
+import { OpenSearchSearchHit } from '../../doc_views/doc_views_types';
+import { DiscoverOptions } from '../discover_options/discover_options';
+
+export interface DiscoverResultsActionBarProps {
+  hits?: number;
+  showResetButton?: boolean;
+  resetQuery(): void;
+  rows?: OpenSearchSearchHit[];
+  isEnhancementsEnabled: boolean;
+}
+
+export const DiscoverResultsActionBar = ({
+  hits,
+  showResetButton = false,
+  resetQuery,
+  rows,
+  isEnhancementsEnabled,
+}: DiscoverResultsActionBarProps) => {
+  return (
+    <EuiFlexGroup
+      direction="row"
+      gutterSize="none"
+      justifyContent="spaceBetween"
+      className="dscResultsActionBar"
+      data-test-subj="dscResultsActionBar"
+    >
+      <EuiFlexItem>
+        <EuiFlexGroup direction="row" gutterSize="none" justifyContent="flexStart">
+          <EuiFlexItem grow={false}>
+            <HitsCounter
+              hits={hits}
+              showResetButton={showResetButton}
+              onResetQuery={resetQuery}
+              rows={rows}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup direction="row" gutterSize="none" justifyContent="flexStart">
+          {/* TODO: We will allow extensions to be added here */}
```

Comment:
Could you elaborate on what types of extensions are planned for this area?

Reply:
it can be just about anything, but here is an example:

![abc](https://github.com/user-attachments/assets/416e40ad-046f-4a5a-8034-88db9db3df51)